src/Datagrid/ProxyQuery.php

```diff
@@ -114,22 +114,19 @@ class ProxyQuery implements ProxyQueryInterface
      */
     private $hints = [];
 
-    /**
-     * @param QueryBuilder $queryBuilder
-     */
-    public function __construct($queryBuilder)
+    public function __construct(QueryBuilder $queryBuilder)
     {
         $this->queryBuilder = $queryBuilder;
         $this->uniqueParameterId = 0;
         $this->entityJoinAliases = [];
     }
 
-    public function __call($name, $args)
+    public function __call(string $name, array $args)
```

Comment:
should we deprecate and remove this method in admin-bundle?

Reply:
And promote `$proxyQuery->getQueryBuilder()->...` instead ?

I think we should stop to use it in the Sonata code. 
I don't know if we should force the developers to migrate all his code too or if we can let this helper.

I you think we should, I'm ok with the deprecation.

Reply:
I prefer being explicit, but maybe it's too much for devs, maybe it's better to keep it for now.